pandas/core/arrays/integer.py

```diff
@@ -496,22 +496,17 @@ def _values_for_argsort(self):
 
     @classmethod
     def _create_comparison_method(cls, op):
+        @CompWrapper(validate_len=True, inst_from_senior_cls=True)
```

Comment:
this changes the behavior to defer to DataFrame as well as Series and Index.  I agree that it _should_ do this eventually, but IIRC there are tests that fail if we start doing it now.